src/env/node/pathMapping/sharedGKDataFolder.ts

```diff
@@ -1,73 +1,134 @@
 import os from 'os';
 import path from 'path';
+import { env } from 'process';
 import { Uri, workspace } from 'vscode';
+import { xdgData } from 'xdg-basedir';
 import { Logger } from '../../../system/logger';
 import { wait } from '../../../system/promise';
 import { getPlatform } from '../platform';
 
-export const sharedGKDataFolder = '.gk';
+/** @deprecated prefer using XDG paths */
+const legacySharedGKDataFolder = path.join(os.homedir(), '.gk');
 
-export async function acquireSharedFolderWriteLock(): Promise<boolean> {
-	const lockFileUri = getSharedLockFileUri();
+class SharedGKDataFolderMapper {
+	private _initPromise: Promise<void> | undefined;
+	constructor(
+		// do soft migration, use new folders only for new users (without existing folders)
```

Comment:
We may want to inform users who have a `.gk` folder, and want to use the new location, like many of the users in the original issue, that they should delete the former. Otherwise GitLens will continue using it.

In fact, they may want to take the contents out of that folder, delete it, and then move it into the new XDG-friendly folder so they don't lose the data.

Reply:
Yeah, probably in the help center and on the issue, but I wouldn't say in the app